src/Orleans.Transactions/InClusterTM/TransactionManager.cs

```diff
@@ -43,8 +45,9 @@ public class TransactionManager : ITransactionManager
         protected readonly ILogger logger;
         private bool IsRunning;
         private Task transactionLogMaintenanceTask;
-
-        public TransactionManager(TransactionLog transactionLog, IOptions<TransactionsConfiguration> configOption, ILoggerFactory loggerFactory, TimeSpan? logMaintenanceInterval = null)
+        private TransactionManagerMetrics metrics;
+        public TransactionManager(TransactionLog transactionLog, IOptions<TransactionsConfiguration> configOption, ILoggerFactory loggerFactory, ITelemetryProducer telemetryProducer,
+            Factory<NodeConfiguration> getNodeConfig, TimeSpan? logMaintenanceInterval = null)
```

Comment:
I suggest getting IStatisticsConfiguration rather than NodeConfig.  We only need the statistics configuration.  This may require registering the NodeConfiguration as an IStatisticsConfiguration  in DI.  Also, not clear why getNodeConfig is a factory function?

Reply:
I would agree if we were not moving toward to option configuration. In my recollection, there should be a StatisticsOption , or some sort , as the new recommended way to configure statistics. And when that happens, this factory func will be replaced by an IOption<StatisticOption> and injected from DI. 

I don't think we should add more reference of old legacy config interfaces to Orleans core now.